### PR TITLE
Feature/Include pushing the different event types

### PR DESCRIPTION
**Pull request by JulissaDantes**
## Description
Allows to have different event codes for Firehose so that users could perform different actions according to whether an event is triggered by a stream creation, a stream anchoring or a stream update from the user.

### CODE REVIEW

File: packages/core/src/feed.ts [hunk 1]
```diff
@@ -1,20 +1,53 @@
-import type { StreamMetadata, StreamState } from '@ceramicnetwork/common'
+import { CommitType, LogEntry, StreamMetadata, StreamState } from '@ceramicnetwork/common'
 import { Subject, type Observable } from 'rxjs'
 import { CommitID } from '@ceramicnetwork/streamid'
 import { StreamUtils } from '@ceramicnetwork/common'
 
+/**
+ * Describes all event types
+ */
+export enum EventType {
```
Comment (stbrody) [inline packages/core/src/feed.ts:9]: I believe the standard names for these from the protocol spec now are Init Events, Data Events, and Time Events.  So presumably that means this enum should have INIT, DATA, and TIME values.

Tagging @oed to confirm.

File: packages/core/src/feed.ts [hunk 2]
```diff
@@ -1,20 +1,53 @@
...
+  CREATE = 0,
+  ANCHOR = 1,
```
Comment (stbrody) [inline packages/core/src/feed.ts:11]: toooooottaal nitpick that really doesn't matter at all, but I feel like I'll have an easier time remembering the values these represent if init/create is 0, data/update is 1, and time/anchor is 2.

File: packages/core/src/feed.ts [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
+ * @param log
+ * @returns EventType
+ */
+function parseCommitType(log: LogEntry[]): EventType {
+  const length = log.length
+
+  if (length == 1) return EventType.CREATE
+
+  switch (log[length - 1].type) {
+    case CommitType.ANCHOR:
```
Comment (stbrody) [inline packages/core/src/feed.ts:26]: why not just use the CommitType enum (and make a change to the names of the values in that enum to match the current naming convention)?  Why have two enums and map between the two of them?

File: packages/core/src/feed.ts [hunk 4]
```diff
@@ -1,20 +1,53 @@
...
+      return EventType.ANCHOR
+    case CommitType.GENESIS:
+      return EventType.CREATE
+    case CommitType.SIGNED:
+      return EventType.UPDATE
+    default:
```
Comment (stbrody) [inline packages/core/src/feed.ts:32]: this worries me, can we throw an error if the log doesn't have the type set properly?  That would probably indicate a bug in our code if that ever happened.

**Review by stbrody — CHANGES_REQUESTED** [review]

**Review by ukstv — APPROVED** [review]

**Pull request merged**